File: client/src/main/java/net/runelite/client/plugins/externals/oneclick/config/Skilling.java
package net.runelite.client.plugins.externals.oneclick.config;

import java.util.List;
import java.util.Objects;
import java.util.Set;
import java.util.stream.Collectors;
import lombok.AllArgsConstructor;
import lombok.Getter;
import net.runelite.client.plugins.externals.oneclick.clickables.Clickable;
import net.runelite.client.plugins.externals.oneclick.clickables.skilling.Birdhouses;
import net.runelite.client.plugins.externals.oneclick.clickables.skilling.Bones;
import net.runelite.client.plugins.externals.oneclick.clickables.skilling.HighAlch;
import net.runelite.client.plugins.externals.oneclick.clickables.skilling.Karams;
import net.runelite.client.plugins.externals.oneclick.clickables.skilling.Runes;
import net.runelite.client.plugins.externals.oneclick.clickables.skilling.Tiaras;
import net.runelite.client.plugins.externals.oneclick.clickables.skilling.farming.FarmingRun;
import net.runelite.client.plugins.externals.oneclick.clickables.skilling.farming.TitheFarm;

@Getter
@AllArgsConstructor
public enum Skilling
{
	BIRDHOUSES(Birdhouses.class),
	BONES(Bones.class),
	FARMING_RUN(FarmingRun.class),
	HIGH_ALCH(HighAlch.class),
	KARAMBWANS(Karams.class),
	RUNES(Runes.class),
	TIARA(Tiaras.class),
	TITHE_FARM(TitheFarm.class),
	;

	private final Class<? extends Clickable> clazz;

	public static List<Clickable> createInstances(Skilling[] skillings)
	{
		Set<Skilling> set = Set.of(skillings);
		return set.stream()
			.filter(Objects::nonNull)
			.map(skilling -> {
				try {
					return skilling.getClazz().newInstance();
				} catch (InstantiationException | IllegalAccessException e) {
					throw new RuntimeException(e);
				}
			})
			.collect(Collectors.toList());
	}
}

File: client/src/main/java/net/runelite/client/plugins/externals/oneclick/config/Skilling.kt
package net.runelite.client.plugins.externals.oneclick.config

import net.runelite.client.plugins.externals.oneclick.clickables.Clickable
import net.runelite.client.plugins.externals.oneclick.clickables.skilling.*
import net.runelite.client.plugins.externals.oneclick.clickables.skilling.farming.FarmingRun
import net.runelite.client.plugins.externals.oneclick.clickables.skilling.farming.TitheFarm
import java.util.*
import java.util.Set
import java.util.stream.Collectors
import kotlin.collections.List

enum class Skilling(val clazz: Class<out Clickable>) {
    BIRDHOUSES(Birdhouses::class.java), BONES(Bones::class.java), FARMING_RUN(FarmingRun::class.java), HIGH_ALCH(
        HighAlch::class.java
    ),
    KARAMBWANS(Karams::class.java), RUNES(Runes::class.java), TIARA(Tiaras::class.java), TITHE_FARM(
        TitheFarm::class.java
    );

    companion object {
        fun createInstances(skillings: Array<Skilling>): List<Clickable> {
            val set = mutableSetOf(*skillings)
            return set.stream()
                .filter { obj: Skilling -> Objects.nonNull(obj) }
                .map { skilling: Skilling ->
                    try {
                        return@map skilling.clazz.newInstance()
                    } catch (e: InstantiationException) {
                        throw RuntimeException(e)
                    } catch (e: IllegalAccessException) {
                        throw RuntimeException(e)
                    }
                }
                .collect(Collectors.toList())
        }
    }
}